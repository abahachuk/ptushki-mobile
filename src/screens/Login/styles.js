import { StyleSheet } from "react-native";

/* eslint  import/prefer-default-export: 0 */
export const styles = StyleSheet.create({
  container: {
    width: "100%",
    flex: 1,
    justifyContent: "space-between",
    paddingRight: 10,
    paddingLeft: 10
  },
  logoImg: {
    width: 300,
    alignSelf: "center"
  },
  headerText: {
    fontSize: 20,
    textAlign: "center",
    paddingBottom: 15
  },
  footer: {
    paddingTop: 20
  }
});
