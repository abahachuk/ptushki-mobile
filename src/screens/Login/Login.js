import React, { useState } from "react";
import PropTypes from "prop-types";
import { View, Text, Image, KeyboardAvoidingView } from "react-native";

/* eslint-disable */
import { Button, Input } from "components";
import { translate } from "../../i18n";
import { makeValidatorEmail, makeValidatorPassword } from "utils/validators";
import { styles } from "./styles";

const logoImg = require("assets/logotype/logotype2x.png");
const infoImg = require("assets/ic_info/ic_info2x.png");
/* eslint-enable */

const Login = props => {
  const { email: emailFromProps, password: passwordFromProps } = props;
  const [email, setEmail] = useState(emailFromProps);
  const [emailError, setEmailError] = useState("");
  const [password, setPassword] = useState(passwordFromProps);
  const [passwordError, setPasswordError] = useState("");

  const onLoginPress = () => {
    props.onLogin({ email, password });
  };
  const onRegisterPress = () => {
    props.onRegister();
  };
  const onPasswordForgot = () => {};
  const validateEmail = makeValidatorEmail(translate("validationError.email"));
  const validatePassword = makeValidatorPassword(
    translate("validationError.password")
  );
  const onTextInputBlur = () => {
    setEmailError(validateEmail(email));
    setPasswordError(validatePassword(password));
  };

  return (
    <KeyboardAvoidingView style={styles.container} enabled>
      <View style={styles.loginContainer}>
        <View style={styles.headerContainer}>
          <View style={styles.infoImgContainer}>
            <Image style={styles.infoImg} source={infoImg} />
          </View>
          <View style={styles.header}>
            <Image
              style={styles.logoImg}
              resizeMode="contain"
              source={logoImg}
            />
            <Text style={styles.headerText}>
              {translate("login.bandingCenter")}
            </Text>
          </View>
        </View>
        <View style={styles.loginForm}>
          <Input
            value={email}
            label={translate("login.email")}
            textContentType="emailAddress"
            onChangeText={setEmail}
            error={emailError}
            onTextInputBlur={onTextInputBlur}
          />
          <Input
            value={password}
            label={translate("login.password")}
            textContentType="password"
            onChangeText={setPassword}
            error={passwordError}
            onTextInputBlur={onTextInputBlur}
            wrapperStyles={[styles.passwordInput]}
          />
          <Button
            caption={translate("login.sign-in")}
            onPress={onLoginPress}
            appearance="Dark"
            style={styles.signInBtn}
          />
        </View>
      </View>
      <View style={styles.footer}>
        <Button
          caption={translate("login.sign-up")}
          onPress={onRegisterPress}
          appearance="Light"
        />
        <Button
          caption={translate("login.forgotPassword")}
          onPress={onPasswordForgot}
          appearance="Borderless"
          style={styles.restorePswBtn}
        />
      </View>
    </KeyboardAvoidingView>
  );
};

Login.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  onLogin: PropTypes.func,
  onRegister: PropTypes.func
};
Login.defaultProps = {
  email: "",
  password: "",
  onLogin: () => {},
  onRegister: () => {}
};

export default Login;
