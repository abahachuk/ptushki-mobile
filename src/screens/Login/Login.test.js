import React from "react";
import renderer from "react-test-renderer";

import Login from "./Login";

describe("Login screen", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<Login />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
