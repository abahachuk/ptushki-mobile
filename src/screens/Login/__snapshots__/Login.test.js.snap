// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "paddingTop": 20,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <Image
            source="\\"ic_info.png\\""
            style={
              Object {
                "height": 22,
                "width": 22,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 10,
            "paddingTop": 10,
          }
        }
      >
        <Image
          resizeMode="contain"
          source="\\"logo.png\\""
          style={
            Object {
              "alignSelf": "center",
              "height": 104,
              "width": 186,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#546e7a",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.1,
              "lineHeight": 18,
              "paddingBottom": 15,
              "paddingTop": 15,
              "textAlign": "center",
            }
          }
        >
          [missing "en.login.bandingCenter" translation]
        </Text>
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View
        style={Array []}
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "borderColor": "#000000",
                  "borderRadius": 4,
                  "borderWidth": 0.5,
                  "height": 56,
                  "paddingLeft": 15,
                  "position": "relative",
                  "width": "100%",
                },
              ],
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.6)",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "left": 15,
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                  "position": "absolute",
                  "top": 15,
                },
              ]
            }
          >
            [missing "en.login.email" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.6)",
                  "fontSize": 16,
                },
                undefined,
              ]
            }
            textContentType="emailAddress"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={Array []}
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "borderColor": "#000000",
                  "borderRadius": 4,
                  "borderWidth": 0.5,
                  "height": 56,
                  "paddingLeft": 15,
                  "position": "relative",
                  "width": "100%",
                },
              ],
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.6)",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "left": 15,
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                  "position": "absolute",
                  "top": 15,
                },
              ]
            }
          >
            [missing "en.login.password" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.6)",
                  "fontSize": 16,
                },
                undefined,
              ]
            }
            textContentType="password"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={
              Array [
                Object {
                  "marginTop": 15,
                },
              ]
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 15,
                  "top": 15,
                },
              ]
            }
          >
            <Image
              source="\\"ic_eye_show.png\\""
              style={
                Array [
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#546e7a",
            "borderRadius": 4,
            "height": 36,
            "marginTop": 20,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#546e7a",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 1.25,
                "lineHeight": 16,
                "paddingTop": 10,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Array [],
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          [missing "en.login.sign-in" translation]
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#546e7a",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 36,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#546e7a",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "fontWeight": "500",
              "letterSpacing": 1.25,
              "lineHeight": 16,
              "paddingTop": 10,
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            Array [],
            undefined,
          ]
        }
      >
        [missing "en.login.sign-up" translation]
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 4,
          "height": 36,
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#546e7a",
              "fontFamily": "Roboto",
              "fontSize": 14,
              "fontWeight": "500",
              "letterSpacing": 1.25,
              "lineHeight": 16,
              "paddingTop": 10,
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            Array [],
            undefined,
          ]
        }
      >
        [missing "en.login.forgotPassword" translation]
      </Text>
    </View>
  </View>
  <View />
</View>
`;
